app/Repositories/Group/GroupRepository.php

```diff
@@ -0,0 +1,507 @@
+                            [
+                                'group_id' => $group->id,
+                                'project_id' => $project->id,
+                                'user_id' => $user->id,
+                                'created_at' => now(),
+                                'updated_at' => now(),
+                            ],
+                        ]);
+                }
+            }
+        }
+    }
+
+    /**
+     * Update Group / Project / User relationship
+     *
+     * @param $group
+     * @param $projects
+     * @param $users
+     */
+    public function updateGroupProjectUser($group, $projects, $users)
+    {
+        $group = $this->model->find($group->id);
+        $auth_user = auth()->user();
+
+        if ($group) {
+            DB::table('group_project_user')
+                ->where('group_id', $group->id)
+                ->delete();
+
+            foreach ($projects as $projectId) {
+                DB::table('group_project_user')
+                    ->insertOrIgnore([
+                        [
+                            'group_id' => $group->id,
+                            'project_id' => $projectId,
+                            'user_id' => $auth_user->id,
+                            'created_at' => now(),
+                            'updated_at' => now(),
+                        ],
+                    ]);
+
+                foreach ($users as $user) {
+                    DB::table('group_project_user')
+                        ->insertOrIgnore([
+                            [
+                                'group_id' => $group->id,
+                                'project_id' => $projectId,
+                                'user_id' => $user->id,
+                                'created_at' => now(),
+                                'updated_at' => now(),
+                            ],
+                        ]);
+                }
+            }
+        }
+    }
+
+    /**
+     * Remove Group / Project / User relationship
+     *
+     * @param $group
+     * @param $user
+     */
+    public function removeGroupProjectUser($group, $user)
+    {
+        $group = $this->model->find($group->id);
+
+        if ($group) {
+            foreach ($group->projects as $project) {
+                DB::table('group_project_user')->where('group_id', $group->id)
+                    ->where('project_id', $project->id)
+                    ->where('user_id', $user->id)
+                    ->delete();
+            }
+        }
+    }
+
+    /**
+     * Remove invited user
+     *
+     * @param $group
+     * @param $email
+     */
+    public function removeInvitedUser($group, $email)
+    {
+        $group = $this->model->find($group->id);
+
+        if ($group) {
+                DB::table('invited_group_users')
+                    ->where('group_id', $group->id)
+                    ->where('invited_email', $email)
+                    ->delete();
+        }
+    }
```

Comment:
Formatting issue; Appears to be double indented .

Reply:
Removed the double indentation in GroupRepository@removeInvitedUser().